Extra Transaction Details for BTC

sag333ar opened the pull request.

sag333ar committed:
loading more details for btc trxns

sag333ar committed:
more details

sag333ar committed:
more changes as per the discussion in meeting

miloridenour commented on src/routes/(authed)/transactions/Table/tr/ContractTr.svelte:
```diff
@@ -22,35 +43,191 @@
+					toAccount = details.payload?.recipient_btc_address ?? 'Unknown';
+				} else if (details.action === 'map') {
+					fromAccount = 'Bitcoin Network';
+					const instruction = details.payload?.tx_data?.instructions?.[0];
```
When parsing the instruction, convert the entire string into URL search params and extract the `deposit_to` key from those, which will allow us to add other "keywork" keys such as `swap_to` in the future.

This whole system was not finding the recipient correctly for my transactions, example tx where it's broken:
https://vsc.techcoderx.com/tx/869bb425ab1a8a92e5654b58949474dbd728c29b

miloridenour commented on src/routes/(authed)/transactions/Table/tr/ContractTr.svelte:
```diff
@@ -61,39 +238,106 @@
+			<p class="error">{error}</p>
+		{:else}
+			<div class="amount section">
+				<h3>Amount</h3>
```
remove this whole `<h3>` line to match the sidebars for hive transactions

miloridenour commented on src/routes/(authed)/transactions/Table/tr/ContractTr.svelte:
```diff
@@ -22,35 +43,191 @@
+					try {
+						payloadContent = JSON.parse(parsed.payload);
+					} catch (e) {
+						console.error('Failed to parse payload JSON string:', parsed.payload, e);
```
There should always be more information provided than this, similar to how it is before. At a minimum, the contract ID, transaction ID, and a link to the block explorer should be provided. 
<img width="382" height="97" alt="image" src="https://github.com/user-attachments/assets/5813953f-fc6a-4ed4-a8bf-ea4753382c16" />

miloridenour reviewed (commented):
Some things to work on, but overall doing a much better job of showing all of the relevant information

sag333ar committed:
Addressed Pull Request comments

sag333ar committed:
Support map, unmap & transfer to BtcMappingTr Screen, and everything else to ContractTr Screen

miloridenour commented on src/routes/(authed)/transactions/Table/tr/BtcMappingTr.svelte:
```diff
@@ -0,0 +1,483 @@
+	};
+});
+
+	const GRAPHQL_QUERY = `query AccHistory ($opts: TransactionFilter) { txns: findTransaction(filterOptions: $opts) { id anchr_height anchr_ts required_auths required_posting_auths nonce status ops { type, index, data } rc_limit ledger { type from to amount asset memo params } ledger_actions { type status to amount asset memo data } output { id index } }}`;
```
Not really sure what you're doing a second query here, but we should never be doing any sort of manual graphql queries outside of the houdini engine. This information should all be available already, and is being passed in as part of the tx prop.

miloridenour commented on src/routes/(authed)/transactions/Table/tr/BtcMappingTr.svelte:
```diff
@@ -0,0 +1,483 @@
+
+			{#if details?.payload?.tx_data && details.action === 'map'}
+				<div class="section">
+					<h3>Deposit TX Data</h3>
```
This is the type of detail we can leave for the block explorer. This doesn't tell the user information they readily need to understand what the transaction was doing, where it was sending money, etc, so we shouldn't include it in Altera. The Altera display interface is designed to be provide a high level abstraction of what is happening, and not display any information that is more related to the protocol than the end user result.

miloridenour commented on src/routes/(authed)/transactions/Table/tr/BtcMappingTr.svelte:
```diff
@@ -0,0 +1,483 @@
+
+			if (!hafRes.ok || !gqlRes.ok) throw new Error('Fetch failed');
+
+			const hafData = await hafRes.json();
```
I have no clue what hafData is, but we should never be querying this endpoint.

miloridenour commented on src/routes/(authed)/transactions/Table/tr/BtcMappingTr.svelte:
```diff
@@ -0,0 +1,483 @@
+		error = '';
+		try {
+			const [hafRes, gqlRes] = await Promise.all([
+				fetch(`https://techcoderx.com/hafah-api/transactions/${tx.id}?include-virtual=true`),
```
Making a manual graphql query, and making it to techcoderx.com/api/v1/graphql instead of the main vsc endpoint https://api.vsc.eco/api/v1/graphql is not idiomatic here. These queries are all redundant, anyway, but if you do need to make queries in the future, make them via houdini which has site-wide configuration to the main endpoint.

miloridenour reviewed (changes requested):
A few things to work on with the graphql, but good progress.

sag333ar committed:
refactor(transactions): migrate to Houdini GraphQL stores and remove HAF API dependencies
- Extract inline GraphQL queries to separate .gql files for auto-generation
- Remove mappingAction.ts and HAF API dependency for action detection
- Optimize BtcMappingTr component data fetching

sag333ar committed:
removed unwanted HAFAF api call

sag333ar committed:
Address Pull request Feedback

sag333ar committed:
Merge branch 'main' of https://github.com/vsc-eco/altera-app into feature/btc-tx-details

miloridenour committed:
query all outputs, consolidate AccHistory into GetTransactions

sag333ar committed:
messages and errors are showing up on the sidebar

sag333ar committed:
Update BTC transaction mapping to support new contract schema

miloridenour commented:
- Absorbed by recolor-btc-support branch
